src/test/java/org/prebid/server/bidder/stroeercore/StroeerCoreBidderTest.java

```diff
@@ -0,0 +1,381 @@
+        final HttpCall<BidRequest> httpCall = createHttpCall(response);
+
+        // when
+        final Result<List<BidderBid>> result = bidder.makeBids(httpCall, BidRequest.builder().build());
+
+        // then
+        final Bid expectedBid1 = Bid.builder()
+                .id("1")
+                .impid("1929")
+                .adm("<div></div>")
+                .price(new BigDecimal("0.30"))
+                .crid("foo")
+                .w(300)
+                .h(600)
+                .build();
+
+        final Bid expectedBid2 = Bid.builder()
+                .id("27")
+                .impid("2010")
+                .adm("<span></span>")
+                .price(new BigDecimal("1.58"))
+                .crid("bar")
+                .w(800)
+                .h(250)
+                .build();
+
+        assertThat(result.getErrors()).isEmpty();
+        assertThat(result.getValue()).containsOnly(BidderBid.of(expectedBid1, BidType.banner, "EUR"),
+                BidderBid.of(expectedBid2, BidType.banner, "EUR"));
+    }
+
+    @Test
+    public void makeBidsShouldReturnErrorIfResponseBodyCouldNotBeParsed() {
+        // given
+        final HttpCall<BidRequest> httpCall = createHttpCall("{}");
+
+        // when
+        final Result<List<BidderBid>> result = bidder.makeBids(httpCall, null);
+
+        // then
+        assertThat(result.getValue()).isEmpty();
+        assertThat(result.getErrors())
+                .hasSize(1)
+                .first()
+                .satisfies(error -> assertThat(error.getType()).isEqualTo(BidderError.Type.bad_server_response))
+                .satisfies(error -> assertThat(error.getMessage()).startsWith("Failed to decode"));
+    }
+
+    @Test
+    public void makeBidsShouldReturnEmptyListIfZeroBids() throws JsonProcessingException {
+        // given
+        final HttpCall<BidRequest> httpCall = createHttpCall(StroeercoreBidResponse.of(emptyList()));
+
+        // when
+        final Result<List<BidderBid>> result = bidder.makeBids(httpCall, null);
+
+        // then
+        assertThat(result.getErrors()).isEmpty();
+        assertThat(result.getValue()).isEmpty();
+    }
+
+    private void assertHttpRequest(final BidRequest expectedBidRequest, final HttpRequest<BidRequest> httpRequest)
+            throws IOException {
+        assertThat(mapper.readValue(httpRequest.getBody(), BidRequest.class)).isEqualTo(expectedBidRequest);
+        assertThat(httpRequest.getMethod()).isEqualTo(HttpMethod.POST);
+        assertThat(httpRequest.getUri()).isEqualTo(ENDPOINT_URL);
+        assertThat(httpRequest.getPayload()).isEqualTo(expectedBidRequest);
+        assertThat(httpRequest.getHeaders())
+                .extracting(Map.Entry::getKey, Map.Entry::getValue)
+                .containsOnly(
+                        tuple("Content-Type", "application/json;charset=utf-8"),
+                        tuple("Accept", "application/json"));
+    }
```

Comment:
It's unit tests and you need to assert only specific test cases. No need to assert whole request. For example in `makeHttpRequestsShouldReturnExpectedRequest` assertion part can be presented like:
```
assertThat(result.getValue())
                .extracting(HttpRequest::getPayload)
                .flatExtracting(BidRequest::getImp)
                .extracting(Imp::getTagid)
                .containsExactly("192848", "abc");
```

Reply:
Made changes here. Is everything ok now?